File: articles/symmetry-conservation-and-noethers-theorem/fortran/conservation_table.f90
! Conservation Table
!
! This Fortran workflow computes energy and angular momentum for
! sample central-force states.

program conservation_table
  implicit none

  integer :: i
  real(8), parameter :: mu = 1.0d0
  real(8), dimension(4) :: x_values
  real(8), dimension(4) :: y_values
  real(8), dimension(4) :: vx_values
  real(8), dimension(4) :: vy_values
  real(8) :: r
  real(8) :: energy
  real(8) :: angular_momentum

  x_values = (/1.0d0, 1.0d0, 2.0d0, 1.0d0/)
  y_values = (/0.0d0, 0.0d0, 0.0d0, 1.0d0/)
  vx_values = (/0.0d0, 0.0d0, 0.0d0, -0.3d0/)
  vy_values = (/0.8d0, 1.0d0, 0.5d0, 0.7d0/)

  print *, "case_id,x,y,vx,vy,energy,angular_momentum_z"

  do i = 1, 4
     r = sqrt(x_values(i)**2 + y_values(i)**2)
     energy = 0.5d0 * (vx_values(i)**2 + vy_values(i)**2) - mu / r
     angular_momentum = x_values(i) * vy_values(i) - y_values(i) * vx_values(i)

     print '(I4,A,F10.5,A,F10.5,A,F10.5,A,F10.5,A,F12.8,A,F12.8)', &
       i, ",", x_values(i), ",", y_values(i), ",", vx_values(i), ",", &
       vy_values(i), ",", energy, ",", angular_momentum
  end do

end program conservation_table

File: articles/symmetry-conservation-and-noethers-theorem/c/conservation_table.c
/*
Conservation Table

This C workflow computes energy and angular momentum for sample
central-force states.
*/

#include <math.h>
#include <stdio.h>

double energy(double mu, double x, double y, double vx, double vy) {
    double r = sqrt(x * x + y * y);
    double kinetic = 0.5 * (vx * vx + vy * vy);
    double potential = -mu / r;
    return kinetic + potential;
}

double angular_momentum_z(double x, double y, double vx, double vy) {
    return x * vy - y * vx;
}

int main(void) {
    const double mu = 1.0;

    double states[4][4] = {
        {1.0, 0.0, 0.0, 0.8},
        {1.0, 0.0, 0.0, 1.0},
        {2.0, 0.0, 0.0, 0.5},
        {1.0, 1.0, -0.3, 0.7}
    };

    printf("case_id,x,y,vx,vy,energy,angular_momentum_z\n");

    for (int i = 0; i < 4; i++) {
        double x = states[i][0];
        double y = states[i][1];
        double vx = states[i][2];
        double vy = states[i][3];

        printf("case_%d,%.8f,%.8f,%.8f,%.8f,%.12f,%.12f\n",
               i + 1,
               x,
               y,
               vx,
               vy,
               energy(mu, x, y, vx, vy),
               angular_momentum_z(x, y, vx, vy));
    }

    return 0;
}
